news/feature-4354.md

```diff
@@ -0,0 +1,19 @@
+`python`: Added support for typed custom options.
+
+This applies for `python` source, `python-fetcher` source, `python` destination,
+`python` parser and `python-http-header` inner destination.
+
+Example config:
+```
+python(
+  class("TestClass")
+  options(
+    "string_option" => "example_string"
+    "bool_option" => True  # supported values are: True, False, yes, no
+    "integer_option" => 123456789
```

Comment:
This PR introduces a little breaking change.
In the case of numbers and previously unquoted strings (true/false, yes/no), the Python-side representation changes.

If it would be difficult to keep the original behavior for old config versions, then I think we should mention it in the NEWS file.

Reply:
Nice catch! I have added a runtime warning and a **Breaking change** sentence to the news file entry.

Reply:
Thanks.